### PR TITLE
docs(website): added new blog post - contributor guidelines

### PR DESCRIPTION
**Pull request by jakepage91**
## 📑 Description
Added new blog post to the website called "Why contributor guidelines matter."

## ✅ Checks
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required

## ℹ Additional Information
Images to be included
community-screenshot-2
community-screenshot

### CODE REVIEW

File: website/blog/2024-04-04-contributor-guidelines/index.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+## Understand your audience
+
+Developers are not a monolithic group of internet dwellers who all care about the same problems, have the same experience, or work the same way. Having said that, there are a few things that are broadly applicable. They want to read less and build more. 🏗️
+
+> Developers care about getting started quickly and it’s on the project maintainer to remove as many obstacles as possible to make that happen. 
+
+How can you reduce the time from when a developer finds your project to when they submit their first PR? 
+
+To get started, contributors need to find an issue to work on. Make sure you have open issues at the ready that are tagged to cater to all potential contributors. If a junior developer wants to learn a new language through your project make sure `good first issues` are easy to find. If more ambitious, senior types, are ready to make an impact, make sure it’s clear which issues you would like them to consider, pointing them to a [current milestone](https://github.com/glasskube/glasskube/milestones) or GitHub [project](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).
```
Comment (christophenne) [inline website/blog/2024-04-04-contributor-guidelines/index.mdx:30]: "at the ready": do you mean just "ready" ?

File: website/blog/2024-04-04-contributor-guidelines/index.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+How can you reduce the time from when a developer finds your project to when they submit their first PR? 
+
+To get started, contributors need to find an issue to work on. Make sure you have open issues at the ready that are tagged to cater to all potential contributors. If a junior developer wants to learn a new language through your project make sure `good first issues` are easy to find. If more ambitious, senior types, are ready to make an impact, make sure it’s clear which issues you would like them to consider, pointing them to a [current milestone](https://github.com/glasskube/glasskube/milestones) or GitHub [project](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects). 
+
+## Define contribution paths
+
+Make clear what kind of paths there are to contribute to. Which areas of the project need help? Leverage GitHub tags to segment each project domain. In our case out project has different components such as a `GUI`, a `CLI` and `Package Controller`, `Docs`, a `UI` and the repo itself.
```
Comment (christophenne) [inline website/blog/2024-04-04-contributor-guidelines/index.mdx:34]: "In our case out project" seems to be a typo

File: website/blog/2024-04-04-contributor-guidelines/index.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+- Handling feedback and iteration.
+- Merging contributions into the main codebase.
+- Addressing best practices for commit messages and keeping forks up-to-date.
+
+
+## Development guidelines
+
+Each open-source project aims to solve a different problem and build on a certain stack that might be new to some contributors. Assume they know nothing about your tool and precisely explain the project’s building blocks.
+
+Focus on informing how the code base is built how each component fits together and describing setup environments.  Ideally, contributors should be able to build local versions of the components they contribute to and validate their work.
```
Comment (christophenne) [inline website/blog/2024-04-04-contributor-guidelines/index.mdx:64]: after "code base is built", a colon would be nice I think

File: website/blog/2024-04-04-contributor-guidelines/index.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+- Addressing best practices for commit messages and keeping forks up-to-date.
+
+
+## Development guidelines
+
+Each open-source project aims to solve a different problem and build on a certain stack that might be new to some contributors. Assume they know nothing about your tool and precisely explain the project’s building blocks.
+
+Focus on informing how the code base is built how each component fits together and describing setup environments.  Ideally, contributors should be able to build local versions of the components they contribute to and validate their work. 
+
+If you only find out during the PR review process that it wasn’t clear to the developer, it’s already too late. 😡
```
Comment (christophenne) [inline website/blog/2024-04-04-contributor-guidelines/index.mdx:66]: Yes, but also not too late to adapt for future contributors. I think that could be mentioned – if someone asks, it's a sign that maybe it's not good enough yet – something like that.

File: website/blog/2024-04-04-contributor-guidelines/index.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+## Testing
```
Comment (christophenne) [inline website/blog/2024-04-04-contributor-guidelines/index.mdx:68]: :100: :)

**Review by christophenne — APPROVED** [review]
Great job :) Couple of smaller comments, I'll approve for now and you can decide what you want to change and merge afterwards. 

Since both commits are for the same blog post, I think it would be nice to squash them to one final commit in the end. But I'll leave that up to you.

File: website/blog/2024-04-04-contributor-guidelines/index.mdx (hunk 3 above)
Comment (jakepage91) [reply website/blog/2024-04-04-contributor-guidelines/index.mdx:64]: a comma right?

File: website/blog/2024-04-04-contributor-guidelines/index.mdx (hunk 4 above)
Comment (jakepage91) [reply website/blog/2024-04-04-contributor-guidelines/index.mdx:66]: Sorry, I don't fully understand what you mean here.

File: website/blog/2024-04-04-contributor-guidelines/index.mdx (hunk 1 above)
Comment (christophenne) [reply website/blog/2024-04-04-contributor-guidelines/index.mdx:30]: Oh I didn't know that, then you can leave it of course.

File: website/blog/2024-04-04-contributor-guidelines/index.mdx (hunk 3 above)
Comment (christophenne) [reply website/blog/2024-04-04-contributor-guidelines/index.mdx:64]: yeah sorry, comma i mean :smiley:

File: website/blog/2024-04-04-contributor-guidelines/index.mdx (hunk 4 above)
Comment (christophenne) [reply website/blog/2024-04-04-contributor-guidelines/index.mdx:66]: I just thought "it's already too late" is a bit harsh. It's not too late, it's just an indicator that something was not clear enough, and should be improved.

Comment (jakepage91) [reply website/blog/2024-04-04-contributor-guidelines/index.mdx:66]: ahhhhh got it, true, let me change it

**Comment by jakepage91** [comment]
Corrected and squashed @christophenne 👍

**Review by christophenne — APPROVED** [review]

**Comment by christophenne** [comment]
> Corrected and squashed @christophenne 👍

Perfect, thanks!

**Pull request merged**